File: 3sem/4lab/Banks/models/BankConfigurations/CreditBankAccountConfiguration.cs
using Banks.Exceptions;

namespace Banks.Models.BankConfigurations;

public record CreditBankAccountConfiguration
{
    public CreditBankAccountConfiguration(decimal unverifiedLimit, decimal creditLimit, decimal commission)
    {
        if (creditLimit < 0)
            throw InvalidBankAccountOperationException.OnInvalidCreditLimit();

        if (creditLimit < 0)
            throw InvalidBankAccountOperationException.OnInvalidCreditLimit();

        if (commission < 0)
            throw InvalidBankAccountOperationException.OnInvalidCommission();

        UnverifiedLimit = unverifiedLimit;
        CreditLimit = creditLimit;
        Commission = commission;
    }

    public decimal UnverifiedLimit { get; }
    public decimal CreditLimit { get; }
    public decimal Commission { get; }
}

File: 4sem/1lab/src/main/java/BanksRemake/Domain/Models/BankConfigurations/CreditBankAccountConfiguration.java
package BanksRemake.Domain.Models.BankConfigurations;

/**
 * A set of parameters characterizing the credit bank account.
 * @param unverifiedLimit maximum transaction amount for an unverified user.
 * @param creditLimit the amount you can withdraw minus.
 * @param commission daily commission for being in the minus.
 */
public record CreditBankAccountConfiguration(double unverifiedLimit,
                                             double creditLimit,
                                             double commission) {
}
